### PR TITLE
Split out whitespace rules into multiple rules.

### PR DESCRIPTION
**Pull request by nuevoalex**
Split the overflow indentation check into its own rule.

### CODE REVIEW

File: test/lib/rules/whitespace/whitespace.js [hunk 1]
```diff
@@ -14,7 +14,12 @@ var addPragma = wrappers.addPragma;
 var userConfig = {
   "custom-rules-filename": null,
   "rules": {
-    "whitespace": true
+	"whitespace": true,
+	"conditionals-whitespace": true,
+	"comma-whitespace": true,
+	"operator-whitespace": true,
+	"semicolon-whitespace": true,
+	"function-whitespace": true,
```
Comment (MicahZoltu) [inline test/lib/rules/whitespace/whitespace.js:22]: I would prefer to be explicit in the configuration and mark the one we aren't using as false.  This way if default changes (e.g., if it gets merged into mainline) configuration is still correct.

**Review by MicahZoltu — APPROVED** [review]
I strongly recommend submitting this upstream.  I would _prefer_ to not have to maintain a fork if possible.

File: test/lib/rules/whitespace/whitespace.js (hunk 1 above)
Comment (nuevoalex) [reply test/lib/rules/whitespace/whitespace.js:22]: These are just configuration passed into the tests since it is testing that these rules behave.

**Comment by nuevoalex** [comment]
I submitted a PR upstream as well with context on why we forked

**Pull request merged**